misc/metrics/src/protocol_stack.rs

```diff
@@ -0,0 +1,13 @@
+use libp2p_core::multiaddr::{Multiaddr,Protocol};
+
+//TODO: remove this file and replace with calls to Multiaddr::protocol_stack
+//  once that lands : https://github.com/multiformats/rust-multiaddr/pull/60
+//  This does imply changing the keys from Vec<String> to Vec<'static str>
+
+fn tag(proto: Protocol) -> Option<String> {
+    format!("{}", proto).split("/").skip(1).next().map(str::to_owned)
+}
+
+pub(crate) fn protocol_stack(ma: &Multiaddr) -> Vec<String> {
```

Comment:
```suggestion
pub fn protocol_stack(ma: &Multiaddr) -> Vec<String> {
```

The entire module is crate-private so we don't need this! :)

Reply:
Can we have a few tests for this please?

Reply:
Honestly I was trying to go minimal here because in my mind the "real" implementation is still [in multiaddr](https://github.com/multiformats/rust-multiaddr/pull/60). But yeah I could put a test here too.

Reply:
Oh I wasn't aware of that PR, sorry about that!

We'd still probably want a test that shows that we construct the correct stack as it is described in the issue. It makes review easier because I can see the output of the function without having to run or brain-compile it :)